BUILD.md

```diff
@@ -216,4 +217,35 @@ Fib/20/lfvm-no-code-cache-12  21.2ms ± 3%  20.7ms ± 3%  -2.17%  (p=0.006 n=9+1
 
 For details on how to use it, please refer to the [Example](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat#hdr-Example) section of the tool's documentation page.
 
-Ideally, pull requests targeting performance improvements should include such a report in their description to document its impact.
\ No newline at end of file
+Ideally, pull requests targeting performance improvements should include such a report in their description to document its impact.
+
+## Code Coverage
+
+Code coverage of the Tosca project is a work in progress. As the different systems get integrated, more features will be added and documented here.
+
+### C++ Code coverage
+
+Current code coverage infrastructure uses GNU gcov system. This is supported in both Gcc and Clang, although Clang target gcov version may not be the installed one. For this reason the current infrastructure is currently enabled for Gcc only. 
+
+Code coverage depends on the following packages:
+- lcov
+- genhtml
+Both commands can be installed in Ubuntu with the command: `apt install lcov`
+
+Code coverage can be triggered by using the following command:
+```
+CC=gcc CXX=g++ make test-cpp-coverage
```

Comment:
I tried to run this command, but I got this error:
```
$ CC=gcc CXX=g++ make test-cpp-coverage
cd cpp ; \
cmake \
	-Bbuild \
	-DCMAKE_BUILD_TYPE="Debug" \
	-DCMAKE_SHARED_LIBRARY_SUFFIX_CXX=.so \
	-DTOSCA_ASSERT="ON" \
	-DTOSCA_COVERAGE="ON" \
	-DTOSCA_ASAN="OFF"; \
cmake --build build --parallel
CMake Error at CMakeLists.txt:28 (message):
  Coverage is only supported with GCC.
```

After a while I realized that the `cmake` build was already configured, and it is ignoring the new build. Running
```
make clean
```
fixed the problem.

Should this be mentioned here somewhere?

Reply:
Sorry, this is a no-brainer to me. CMake cannot change toolchain once configured. I'll write it down.